### PR TITLE
Can now add members/admins to group

### PR DESCRIPTION
**Pull request by shaneporter**
`editor.allTheUsers` _was_ used to initialise the users. Although that sounded about right, its value was undefined, and there was no code to get the users.

I have added a new scope level function to get the users. This wasn't enough, however, since I couldn't get things to work until I converted the names to selectize compatible format.

Becuase both selectize directives use the scope level `allUsers` member it works for both admins and members.

### CODE REVIEW

File: htdocs/js/ui/notebook_protection_controller.js [hunk 1]
```diff
@@ -94,6 +97,24 @@ define(['angular'], function(angular) {
...
+                .then(function(data) {
+
+                    $scope.$evalAsync(function() {
+                        $scope.allUsers = _.map(data, function(user) { return { text : user, value : user }; });
+                    });
+
+                    //timeout to ensure evalAsync has finished
+                    $timeout(function() {
+                        logger.clear();
+                        resolve();
```
Comment (gordonwoodhull) [inline htdocs/js/ui/notebook_protection_controller.js]: Hmm, is async-call-then-timeout-resolve really the right pattern? Seems like we should just resolve once the asynchronous call is done.

Comment (gordonwoodhull) [inline htdocs/js/ui/notebook_protection_controller.js]: Just a question, I'm going to merge this.

**Comment by gordonwoodhull** [comment]
Ah, okay, this makes sense now - thanks for the explanation. 

Yes, I suggested `allTheUsers` to the previous author of the discovery page as a temporary hack. You removed it in a1f775a01d4 as part of simplifying `load_everything`, which was a good call. It's surprising that it failed so quietly but I'm glad to remember how this used to work.

File: htdocs/js/ui/notebook_protection_controller.js (hunk 1 above)
Comment (shaneporter) [inline htdocs/js/ui/notebook_protection_controller.js]: It's smells a bit doesn't it? Having said that, I took the easier option, and duplicated the issue from other areas of the code. Perhaps it's something that can be looked at in the future.

Comment (gordonwoodhull) [inline htdocs/js/ui/notebook_protection_controller.js]: #2114

**Comment by gordonwoodhull** [comment]
Rebased and merged. Thanks @shaneporter!

**Pull request closed without merging**